File: programas/listasanidadas_matrices.py
#título          :listas anidadas.py
#descripción     :Ejemplos listas anidadas y matrices.
#autor           :Benjamín Valdés
#python_version  :3.5.2  
#===============================================================

# este programa de ejemplo contiene ejercicios de 
# listas anidadas y matrices

def crea_lista_anidada(exterior, interior):

    """ crea una lista anidada de 2 niveles """
    
    lista_exterior = []
    
    i = 0
    while(i < exterior):
        lista_interior = []
        
        j = 0
        while(j < interior):
            lista_interior.append(int(input()))
            j = j + 1
        
        lista_exterior.append(lista_interior)
        i = i + 1
       
    return lista_exterior

    
def suma_diagonal(matriz):
 
    """suma la diagonal de una matriz"""
 
    if len(matriz) != len(matriz[0]):
        return False 
        
    sum = 0
    i = 0
    for line in matriz:
        
        j = 0
        for number in line:
            if i == j:
                sum = sum + number
            j = j + 1
        
        i = i + 1
    return sum
    

ren = int(input())
col = int(input())
mat = crea_lista_anidada(ren,col)
print(mat)
print(suma_diagonal(mat))

File: programas/listasanidadas_matrices.cpp
/*
título          :matrices.cpp
descripción     :Ejercicios sobre matrices.
autor           :Benjamín Valdés
c++ version  	:11
===============================================================

 ejemplo de trabajo con listas en c++
*/

#include<iostream>

using namespace	std;

void llenaMat(int mat[100][100], int exterior, int interior){
	
	/* recorro solo hasta tam, que es hasta 
	donde pienso llenar el arreglo */
	
	for(int i = 0; i < exterior; i++)
		for (int j = 0; j < interior; j++)
			cin >> mat[i][j];
	
}


int sumaDiagonal(int mat[100][100], int exterior, int interior){
	
	int sum = 0;
	
	/* imprimo solo hasta tam que es hasta
	donde llené, lo demás es basura*/
	
	for(int i = 0; i < exterior; i++){
		for (int j = 0; j < interior; j++){
			if (i == j){
				sum = sum + mat[i][j];
			}
		}
	}
	
	return sum;
}


void printMat(int mat[100][100], int exterior, int interior){
	
	cout << "\n";
	for(int i = 0; i < exterior; i++){
		for (int j = 0; j < interior; j++)
			cout << mat[i][j] << " ";
		cout << "\n";
	}
}


int main(){
	
	
	/* Creo un arreglo en c++ con espacio 
	sobrado para lo que quiero guardar */
	
	int matriz[100][100]; 
	
	
	/* Uso las variables exterior e interior 
	para saber hasta que parte de los arregloa 
	voy a usar */
	
	int exterior, interior;
	
	cin >> exterior >> interior;
	
	llenaMat(matriz, exterior, interior);

	printMat(matriz, exterior, interior);

	cout << sumaDiagonal(matriz, exterior, interior);
}